iothub/service/src/Registry/DevicesClient.cs

```diff
@@ -226,6 +226,7 @@ public virtual async Task DeleteAsync(string deviceId, CancellationToken cancell
             try
             {
                 using HttpRequestMessage request = _httpRequestMessageFactory.CreateRequest(HttpMethod.Delete, GetRequestUri(deviceId), _credentialProvider);
+                HttpMessageHelper.ConditionallyInsertETag(request, ETag.All, false);
```

Comment:
This call fails if I don't include this.

Interestingly enough, I can't pass null, as ETag is not nullable. So the code in this method which checks for null is wrong (or we need to make the param nullable.

Thoughts @timtay-microsoft ?

Reply:
For this API, there is never an ETag since the user is only giving us the device Id. There is no reason to call ```HttpMessageHelper.ConditionallyInsertETag``` because of that.

Reply:
Why are you replacing what was here? Calling the overload with a new ```Device``` instance with an empty etag seemed fine to me

Reply:
If I don't make that call, it fails.

Reply:
Because the device one ran through extra code this overload didn't need.

Reply:
Except, apparently, adding the ETag.